### ISSUE TITLE
Fix warnings while building documentation

### ISSUE THREAD

**Issue by neachdainn** [adeschamps/lcm-rust#34]
Currently, building the documentation results in the following warnings:

```
WARNING: documentation for this crate may be rendered differently using the new Pulldown renderer.
    See https://github.com/rust-lang/rust/issues/44229 for details.
WARNING: rendering difference in `From the [LCM Homepage](http://lcm-proj.github.io/):`
   --> lcm/src/lib.rs:1:0
    /html[0]/body[1]/blockquote[1] Unexpected element `p`: found: `<p>LCM is a set of libraries and tools for message ... nd data marshalling,`
WARNING: rendering difference in `From the [LCM Homepage](http://lcm-proj.github.io/):`
   --> lcm/src/lib.rs:1:0
    /html[0]/body[1]/p[2] Text differs:
        expected: `LCM is a set of libraries and tools for message pa ... nd data marshalling,`
        found:    `This crate provides a Rust implementation of the L ...  and code generator.`
WARNING: rendering difference in `From the [LCM Homepage](http://lcm-proj.github.io/):`
   --> lcm/src/lib.rs:1:0
    /html[0]/body[1]/p[2] Unexpected element `code`: found: `<code>lcm-gen</code>`
WARNING: rendering difference in `From the [LCM Homepage](http://lcm-proj.github.io/):`
   --> lcm/src/lib.rs:1:0
    /html[0]/body[1]/p[2] One element is missing: expected: ` crate for generating message types from a specification file.`
WARNING: rendering difference in `From the [LCM Homepage](http://lcm-proj.github.io/):`
   --> lcm/src/lib.rs:1:0
    /html[0]/body[1] One element is missing: expected: `p`
WARNING: rendering difference in `Subscribes a callback to a particular topic.`
   --> lcm/src/lcm/mod.rs:90:4
    /html[0]/body[1]/p[1] Text differs:
        expected: `...^' and '$'.`
        found:    `...`
WARNING: rendering difference in `Subscribes a callback to a particular topic.`
   --> lcm/src/lcm/mod.rs:90:4
    /html[0]/body[1]/p[1] Unexpected element `sup`: found: `<sup>'</sup>`
WARNING: rendering difference in `Subscribes a callback to a particular topic.`
   --> lcm/src/lcm/mod.rs:90:4
    /html[0]/body[1]/p[1] One element is missing: expected: ` and '$'.`
```

**Comment by adeschamps** [comment]
I've seen warnings like these when building the docs for lots of crates. I thought I heard that rustdoc is undergoing some changes in its markdown renderer. Is there anything we can do about this, other than wait for new versions of rustdoc?

**Comment by neachdainn** [comment]
I asked on IRC about this and it seemed that the consensus was that there changes have already happened and these warnings need to be fixed. Or, at the very least, fixing them now wont hurt.

**Comment by adeschamps** [comment]
Fixed in #38
